pulser-core/pulser/sequence.py

```diff
@@ -1428,11 +1438,18 @@ def _phase_shift_non_parametrized(
     def _phase_shift(
         self,
         phi: Union[float, Parametrized],
-        *targets: Union[QubitId, Parametrized],
+        *targets: QubitId,
         basis: str,
     ) -> None:
-        self._precheck_phase_shift(*targets, basis=basis)
-        if not self.is_parametrized():
+        self._check_basis(basis)
+
+        if self.is_parametrized():
+            for t in targets:
+                if t not in self._qids and not isinstance(t, Parametrized):
+                    raise ValueError(
+                        "All non-variable targets must belong to the register."
+                    )
```

Comment:
Same here, this is allowing variable targets

Reply:
Got it, thanks for the remark!